Client/HttpCookieExtension.swift

```diff
@@ -0,0 +1,84 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import WebKit
+import Shared
+
+private let log = Logger.browserLogger
+
+public extension HTTPCookie {
+    
+    typealias Success = Bool
+    class func saveToDisk(_ filename: String = "CookiesData.json", completion: ((Success) -> Void)? = nil) {
+        let cookieStore = WKWebsiteDataStore.default().httpCookieStore
+        
+        /* For reason unkown the callback to getAllCookies is not called, when the save is done from Settings.
+         A possibility is https://bugs.webkit.org/show_bug.cgi?id=188242
+         Even with the issue being fixed it still sometimes doesn't work.
+         The network process is in suspened maybe?
+         
+         And for some reason fetch cookie records preemptively guarantees this works.
+         Best guess is that fetching records brings network process to active.
+         
+         Same applies to setting cookies back below.
+         */
+        WKWebsiteDataStore.default().fetchDataRecords(ofTypes: [WKWebsiteDataTypeCookies]) { (_) in}
+        cookieStore.getAllCookies { (cookies) in
+            let baseDir = NSSearchPathForDirectoriesInDomains(.documentDirectory, .userDomainMask, true)[0]
```

Comment:
This will crash if there no `.documentDirectory`, I know it's not going to happen but we can make it more safely there.

I suggest using `URL` mechanism, also see my comment below
Use something like
 `guard let documentsDir = urls(for: .documentDirectory, in: .userDomainMask).first else { return nil }`  
to get URL of documents dir.

Even better will be creating a helper method for it, since we use it in few places
so
`var documentDirectoryUrl: URL? { return urls(for: .documentDirectory, in: .userDomainMask).first }`

Reply:
We cant work with URL as all the methods use Strings and all the getters return strings. Going ahead with putting a guard to make sure it doesn't crash.